### PR TITLE
Fix multi-module analysis

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeEditorFileAction.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public void actionPerformed(AnActionEvent e) {
     VirtualFile selectedFile = SonarLintUtils.get(SonarLintAppUtils.class).getSelectedFile(project);
 
     if (selectedFile == null) {
-      SonarLintUtils.get(project, SonarLintConsole.class).error("No files for analysis");
+      SonarLintUtils.get(project, SonarLintConsole.class).error("No filesPerModule for analysis");
```
Comment (janos-ss) [inline src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeEditorFileAction.java]: oops ;-)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public AnalysisResults analyzeModule(Module module, Collection<VirtualFile> file
       console.info("No analysis configurator found");
     }
 
-    // configure files
+    // configure filesPerModule
```
Comment (janos-ss) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java]: probably unintended replace...

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java [hunk 3]
```diff
@@ -21,40 +21,51 @@
...
+    this(files, trigger, null);
+  }
+
+  SonarLintJob(Map<Module, Collection<VirtualFile>> files, TriggerType trigger, @Nullable AnalysisCallback callback) {
     this.callback = callback;
-    Preconditions.checkNotNull(m);
+    Preconditions.checkNotNull(files);
     Preconditions.checkNotNull(trigger);
-    Preconditions.checkArgument(!files.isEmpty(), "List of files is empty");
+    Preconditions.checkArgument(!files.isEmpty(), "List of filesPerModule is empty");
```
Comment (janos-ss) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java]: unintended replace?

File: src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java [hunk 4]
```diff
@@ -73,45 +73,42 @@ public void submitFilesModal(Collection<VirtualFile> files, TriggerType trigger)
   }
 
   public void submitFilesModal(Collection<VirtualFile> files, TriggerType trigger, @Nullable AnalysisCallback callback) {
-    Multimap<Module, VirtualFile> filesByModule = filterAndgetByModule(files, false);
+    Map<Module, Collection<VirtualFile>> filesByModule = filterAndgetByModule(files, false);
 
     if (!filesByModule.isEmpty()) {
       console.debug("Trigger: " + trigger);
 
       for (Module m : filesByModule.keySet()) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java]: MINOR Remove this unused "m" local variable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1481)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintTask.java [hunk 5]
```diff
@@ -53,11 +61,13 @@ public SonarLintTask(IssueProcessor processor, SonarLintJob job, boolean backgro
...
    */
   protected SonarLintTask(IssueProcessor processor, SonarLintJob job, boolean modal, boolean background) {
-    super(job.module().getProject(), "SonarLint Analysis", true);
+    super(job.project(), "SonarLint Analysis", true);
     this.processor = processor;
     this.job = job;
     this.modal = modal;
     this.startInBackground = background;
+    this.console = SonarLintConsole.get(job.project());
+    ;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintTask.java]: MINOR Remove this empty statement. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java [hunk 6]
```diff
@@ -21,40 +21,51 @@
...
+import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java]: MINOR Remove this unused import 'java.util.stream.Stream'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java [hunk 7]
```diff
@@ -21,40 +21,51 @@
...
 import com.google.common.base.Preconditions;
 import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintJob.java]: MINOR Remove this unused import 'java.util.Set'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 1 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
No ticket associated?

**Comment by dbmeneses** [comment]
@henryju You're right, I will reopen and associate a ticket

**Pull request merged**